### PR TITLE
Transformers on galleries

### PR DESCRIPTION
**Pull request by NiklasBr**

### CODE REVIEW

File: src/GraphQL/FieldConfigGenerator/Helper/ImageGallery.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ public function resolve($value = null, $args = [], $context = [], ResolveInfo $r
                     }
 
                     if ($image instanceof Asset) {
-                        $data = new \ArrayObject();
+                        $data = new \ArrayObject(['data' => null]);
```
Comment (weisswurstkanone) [inline src/GraphQL/FieldConfigGenerator/Helper/ImageGallery.php]: Please switch to ElementDescriptor.
see https://github.com/pimcore/data-hub/issues/15

File: src/GraphQL/Resolver/HotspotType.php [hunk 2]
```diff
@@ -34,13 +36,36 @@ class HotspotType
      */
     public function resolveImage($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
     {
+        if ($args && $args['defaultLanguage']) {
+            $localeService = \Pimcore::getContainer()->get('pimcore.locale');
```
Comment (weisswurstkanone) [inline src/GraphQL/Resolver/HotspotType.php]: please get rid of Pimcore::getContainer()

see https://github.com/pimcore/data-hub/issues/7 to get an idea

File: src/GraphQL/Resolver/HotspotType.php [hunk 3]
```diff
@@ -34,13 +36,36 @@ class HotspotType
...
+                throw new \Exception('not allowed to view asset ' . $asset->getFullPath());
+            } else {
+                return null;
+            }
+        }
+
+        $data = new \ArrayObject(['data' => null]);
+        $data->setFlags(\ArrayObject::STD_PROP_LIST | \ArrayObject::ARRAY_AS_PROPS);
+
+        $fieldHelper = \Pimcore::getContainer()->get('pimcore.datahub.graphql.fieldhelper.asset');
```
Comment (weisswurstkanone) [inline src/GraphQL/Resolver/HotspotType.php]: same here

**Comment by NiklasBr** [comment]
Thanks for the quick reviews :) I have merged your recent changes but not yet pushed them to GitHub

**Comment by NiklasBr** [comment]
Output:
Skärmbild (147)

The transformer will turn into a list of strings if it receives a gallery rather than an asset field:

Skärmbild (148)

I understand that it might seem a bit hacky, especially the `getResolverAttribute()` part  :)

**Comment by weisswurstkanone** [comment]
thanks a lot!

**Pull request merged**